backend/src/layer/data/database/auth.rs

```diff
@@ -0,0 +1,56 @@
+//! Module responsible for handling database requests for the authentication process.
+
+use async_trait::async_trait;
+use sqlx::{Pool, Postgres};
+
+use crate::interface::persistent_data::{model::ApiKey, AuthDataAccess, Result};
+
+/// Class implementing all database requests arising from graphql manipulations.
+#[derive(Debug)]
+pub struct PersistentAuthData {
+    pub(super) pool: Pool<Postgres>,
+}
+
```

Comment:
Do we want a timestamp column for the api_keys? Or is their lifetime not that important? Idk the auth process that well, if this makes sense...